File: src/main/java/io/github/dqualizer/dqlang/archive/k6configurationrunner/dqlang/request/Request.java
package io.github.dqualizer.dqlang.archive.k6configurationrunner.dqlang.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.archive.k6adapter.dqlang.k6.request.Checks;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Map;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Request {

    private String type;
    private String path;
    @JsonProperty("path_variables")
    private Map<String, String> pathVariables;
    @JsonProperty("query_params")
    private Map<String, String> queryParams;
    private Map<String, String> params;
    private Map<String, String> payload;
    private Checks checks;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/archive/k6configurationrunner/dqlang/request/Request.kt
package io.github.dqualizer.dqlang.archive.k6configurationrunner.dqlang.request

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.archive.k6adapter.dqlang.k6.request.Checks
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.ToString

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
class Request {
    val type: String? = null
    val path: String? = null

    @JsonProperty("path_variables")
    val pathVariables: Map<String, String>? = null

    @JsonProperty("query_params")
    val queryParams: Map<String, String>? = null
    val params: Map<String, String>? = null
    val payload: Map<String, String>? = null
    val checks: io.github.dqualizer.dqlang.archive.k6adapter.dqlang.k6.request.Checks? = null
}
